components/list/list-item-drag-handle.js

```diff
@@ -0,0 +1,196 @@
+
+export const dragActions = Object.freeze({
+	first: 'last',
+	end: 'first',
+	up: 'up',
+	down: 'down',
+	active: 'keyboard-active',
+	save: 'keyboard-deactivate-save',
+	cancel: 'keyboard-deactivate-cancel',
+	nextElement: 'next-element',
+	previousElement: 'previous-element'
+});
+
+class listItemDragHandle extends LitElement {
+
+	static get properties() {
+		return {
+			keyboardActive: { type: Boolean, attribute: 'keyboard-active', reflect: true },
+			text: { type: String }
+		};
+	}
+
+	static get styles() {
+		return [ buttonStyles, css`
+			:host {
+				align-items: center;
+				display: flex;
+				margin: 0.25rem;
+			}
+			:host([hidden]) {
+				display: none;
+			}
+			.d2l-list-item-drag-handle-dragger-button,
+			.d2l-list-item-drag-handle-keyboard-button {
+				display: grid;
+				grid-auto-columns: auto;
+				grid-auto-rows: 1fr 1fr;
+				margin: 0;
+				min-height: 1.8rem;
+				padding: 0;
+				width: 0.9rem;
+			}
+			/* Firefox includes a hidden border which messes up button dimensions */
+			button::-moz-focus-inner {
+				border: 0;
+			}
+			.d2l-list-item-drag-handle-dragger-button {
+				background-color: unset;
+			}
+			button,
+			button[disabled]:hover,
+			button[disabled]:focus {
+				background-color: var(--d2l-color-gypsum);
+				color: var(--d2l-color-ferrite);
+			}
+			button:hover,
+			button:focus {
+				background-color: var(--d2l-color-mica);
+			}
+			button[disabled] {
+				cursor: default;
+				opacity: 0.5;
+			}
+		`];
+	}
+
+	constructor() {
+		super();
+		this.keyboardActive = false;
+	}
+
+	focus() {
+		const node = getFirstFocusableDescendant(this);
+		if (node) {
+			node.focus();
+		}
+	}
+
+	render() {
+		return html`${this.keyboardActive ? this.renderKeyboardDragging() : this.renderDragger()}`;
+	}
+
+	renderDragger() {
+		return html`
+			<button class="d2l-list-item-drag-handle-dragger-button" @click="${this._handleDeactiveKeyboard}" @keydown="${this._handleDeactiveKeyboard}" aria-label="${this.text}">
+				<d2l-icon icon="tier1:dragger"></d2l-icon>
+			</button>
+		`;
+	}
+
+	renderKeyboardDragging() {
+		return html`
+			<button class="d2l-list-item-drag-handle-keyboard-button" @blur="${this._onBlur}" @keydown="${this._handleActiveKeyboard}" aria-label="${this.text}">
+				<d2l-icon class="sub-icon" icon="tier1:arrow-toggle-up" @click="${this._dispatchActionUp}"></d2l-icon>
+				<d2l-icon class="sub-icon" icon="tier1:arrow-toggle-down" @click="${this._dispatchActionDown}"></d2l-icon>
+			</button>
+		`;
+	}
+
+	_dispatchAction(action) {
+		if (!action) return;
+		this.dispatchEvent(new CustomEvent('d2l-list-item-drag-handle-action', {
+			detail: { action },
+			bubbles: false
+		}));
+	}
+
+	_dispatchActionDown() {
+		this._dispatchAction(dragActions.down);
+	}
+
+	_dispatchActionUp() {
+		this._dispatchAction(dragActions.up);
+	}
+
+	_onBlur() {
+		this.keyboardActive = false;
+		if (!this._tabbing) {
+			this._dispatchAction(dragActions.save);
+		}
+	}
+
+	_handleActiveKeyboard(e) {
```

Comment:
This handler, for a native `button` expects keys to be pressed that users would not expect for that element.  I will am interested in how this will be communicated.  I think this is what you might have eluded to in our recent Slack conv...?

Reply:
Yeah, so when people got to this part they were actually pretty good about using the arrow keys. Somethings like `tab` and `shift+tab` were never touched. Also, home and end weren't used either but I figured those would be useful for longer lists.

Yeah, in general Abbie is looking into the best way we can communicate the uses. Probably going to go with a tooltip but waiting on the designs.